src/client/http_client.rs

```diff
@@ -664,23 +665,53 @@ impl QueryRequest {
                 scope,
                 timeout,
                 transaction,
+                parameters,
             } => {
                 let mut json = HashMap::new();
                 if let Some(scope) = scope {
                     let ctx = format!("`default`:`{}`.`{}`", scope.0, scope.1);
-                    json.insert("query_context", ctx);
+                    json.insert("query_context".to_string(), serde_json::Value::String(ctx));
                 }
 
-                json.insert("statement", statement.to_string());
-                json.insert("timeout", timeout.to_string());
+                json.insert(
+                    "statement".to_string(),
+                    serde_json::Value::String(statement.to_string()),
+                );
+                json.insert(
+                    "timeout".to_string(),
+                    serde_json::Value::String(timeout.to_string()),
+                );
                 if let Some(txn) = transaction {
                     if let Some(t) = txn.tx_timeout.clone() {
-                        json.insert("txtimeout", format!("{}ms", t.as_millis()));
+                        json.insert(
+                            "txtimeout".to_string(),
+                            serde_json::Value::String(format!("{}ms", t.as_millis())),
+                        );
                     }
                     if let Some(id) = txn.tx_id.clone() {
-                        json.insert("txid", id);
+                        json.insert("txid".to_string(), serde_json::Value::String(id));
+                    }
+                }
+
+                if let Some(params) = parameters {
+                    match params {
+                        serde_json::Value::Array(_) => {
+                            json.insert("args".to_string(), params.clone());
+                        }
+                        serde_json::Value::Object(map) => {
+                            for (k, v) in map.iter() {
+                                let key = if k.starts_with("$") {
+                                    k.clone()
+                                } else {
+                                    format!("${}", *k)
```

Comment:
I still think we should convert this to &str rather than everything else to string. Do you disagree?

Reply:
I think that we need to have the HashMap be <String, serde_json::Value> so that we can correctly format the positional parameters above. If we have it as a <String, string> then the payload ends up with:
```
args: "{"LAX", 13000}",  - the whole thing is a string
```
But with the <String, Value> map we get what we need where the array is printed as an array.

Reply:
I don't think the compiler will let me. We get `borrowed value does not live long enough` since key leaves scope after the if, but we still use the reference in the json, so I think it needs to be a String.